sqlair.go

```diff
@@ -464,24 +468,34 @@ func (db *DB) Begin(ctx context.Context, opts *TXOptions) (*TX, error) {
 	if ctx == nil {
 		ctx = context.Background()
 	}
-	sqltx, err := db.sqldb.BeginTx(ctx, opts.plainTXOptions())
-	return &TX{sqltx: sqltx, db: db}, err
+	sqlconn, err := db.sqldb.Conn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	sqltx, err := sqlconn.BeginTx(ctx, opts.plainTXOptions())
```

Comment:
The error needs to be checked and nil returned on issues.

Reply:
Changed.